src/EnergyPlus/ZonePlenum.cc

```diff
@@ -1138,22 +1161,19 @@ namespace ZonePlenum {
 				}
 				Node( ZoneNode ).CO2 = Node( OutletNode ).CO2;
 			} else {
-				Node( OutletNode ).CO2 = Node( InletNode ).CO2;
-				Node( ZoneNode ).CO2 = Node( InletNode ).CO2;
+				Node( OutletNode ).CO2 = Node( ZoneNode ).CO2;
```

Comment:
@Myoldmopar This change was a little more subtle.  There is code in ZoneContaminantPredictorCorrector to calculate the CO2 and GenContam concentrations for the plenum zone.  Originally I proposed doing only the else block, but @lgu1234  suggested that the plenum object should update the CO2 and GenContam concentrations here when there is flow, as well as in the ZoneContaminantPredictorCorrector.